Yoast/wordpress-seo
# Create separate tabs for video and news in metabox

## Summary

This PR can be summarized in the following changelog entry:

* [not-userfacing] Introduces a (private) filter for adding metabox tabs: `yoast_free_additional_metabox_sections`.

## Relevant technical choices:

* Add a filter `yoast_free_additional_metabox_sections` that allows for adding additional tabs. This will be used for news and video to render their sections inside their own additional tab.
* Move tests and test doubles to match the folder structure.
* Add namespaces to test doubles.
* Clean/fix some test file and class names.
* Considered a security concern regarding the `link_content` and `content` not being sanitized. However, they are only accessible through a plugin. The solution of `wp_kses_post` was tried, but this strips `input` fields, which is no good.

## Test instructions

This PR can be tested by following these steps:

* Enable the `video` and `news` add-ons and switch them to the same branch name: `13073-create-separate-tabs-for-video-and-news-in-metabox` ([Video](https://github.com/Yoast/wpseo-video/pull/700) and [News](https://github.com/Yoast/wpseo-news/pull/518)).
* The `Addons` tab should be gone and instead, each add-on should have their own tab.
* This is the case for posts, pages and custom posts. As long as you have this turned on in the add-on settings.
  * Video has different content depending on whether you added a video in the content. So test both situations.
* The padding of the tab section is added by the CSS of the add-on itself. It should be 16px, not touching the borders.
  * This should only load in on actual metabox pages (`post` and `post-new`).

## Screenshot
<img width="624" alt="Screenshot 2019-07-30 at 11 16 22" src="https://user-images.githubusercontent.com/5389513/62117022-7e5a8200-b2bb-11e9-8a03-85c3ef0c3493.png">

## UI changes

## Documentation

## Quality assurance


Fixes #13073
